test/irmin-pack/test_pack_version_bump.ml

```diff
@@ -0,0 +1,169 @@
+(** These tests for issue #1658, which follows PR #1655 which
+   introduces a store version bump from V1 to V2.
+
+There are 3 tests:
+
+- test_RO_no_version_bump: open a V1 store RO mode, the version should
+  remain V1
+
+- test_RW_no_version_bump: open a V1 store RW mode, but don't attempt
+  to mutate, version should remain V1
+
+- test_RW_version_bump: open a V1 store RW mode, change something,
+  version should bump to V2
+
+*)
+
+
+open! Import
+open Common
+
+let src = Logs.Src.create "tests.version_bump" ~doc:"Test pack version bump"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Util = struct
+
+  (** Following are generic utils *)
+
+  let tmp_dir () = Filename.temp_file "test_pack_version_bump_" ""
+
+  (** Make a directory; dir is assumed to not exist; parent is assumed to exist *)
+  let mkdir p =
+    assert(not (Sys.file_exists p));
+    Filename.quote_command "mkdir" [p] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Command `%s' exited with non-zero code %d\n" cmd n
+
+
+  (** Copy src to dst; dst is assumed to not exist *)
+  let copy_dir src dst =
+    assert(not (Sys.file_exists dst));
+    (* don't check if it is empty; perhaps we should *)
+    Filename.quote_command "cp" ["-R";src;dst] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Failed to set up the test env; command `%s' exited \
+                    with non-zero code %d\n" cmd n
+
+  (** More specific utils from here *)
+
+  let v1_store_archive_dir = "test/irmin-pack/data/version_1"
+
+  (** Check we are in the location we expect, the project root. See
+     test_existing_stores.goto_project_root for alternative. *)
```

Comment:
The tests are failing in CI here:

```
Command [5] exited with code 2:
$ (cd _build/default/test/irmin-pack && ./test.exe -q --color=always)
Fatal error: exception (Failure
  "Couldn't find dir \"test/irmin-pack/data/version_1\" from cwd \".\"\n")
        test alias test/irmin-bench/runtest
```

The issue is that the current working directory varies according to how Dune is invoked:
- `dune runtest` ⇒ `getcwd` is the parent directory of the compiled test binary (within `_build`);
- `dune exec` ⇒ `getcwd` inherited from the parent process (often the project root).

The approach taken in `test_existing_stores.ml` tries to support both cases by normalising. Can we apply that here too? (Ideally by putting that function somewhere it can be consumed by both modules.)